src/condor_dagman/dagman_main.cpp

```diff
@@ -1364,28 +1367,21 @@ void main_pre_dc_init (int, char*[]) {
 	_setmaxstdio(2048);
 #endif
 
-	// Get the current directory
-	std::string currentDir = "";
-	if (condor_getcwd(currentDir)) {
-		currentDir += DIR_DELIM_STRING;
+	// Convert the DAGMan log file name to an absolute path if it's not one already
+	std::string fullLogFile;
+	const char* logFile = GetEnv("_CONDOR_DAGMAN_LOG");
+	if (logFile) {
+		if ( ! fullpath(logFile)) {
+			fullLogFile = dagman.workingDir + DIR_DELIM_STRING + logFile;
+			SetEnv("_CONDOR_DAGMAN_LOG", fullLogFile.c_str());
+		} else { fullLogFile = logFile; }
 	} else {
-		debug_printf(DEBUG_NORMAL, "ERROR: unable to get cwd: %d, %s\n", errno, strerror(errno));
-	}
-
-	// Convert the DAGMan log file name to an absolute path if it's
-	// not one already
-	std::string newLogFile;
-	const char*	logFile = GetEnv("_CONDOR_DAGMAN_LOG");
-	if (logFile && !fullpath(logFile)) {
-		newLogFile = currentDir + logFile;
-		SetEnv("_CONDOR_DAGMAN_LOG", newLogFile.c_str());
+		fullLogFile = dagman.workingDir + DIR_DELIM_STRING + "default.dagman.out";
```

Comment:
dirscat

Reply:
`dirscat()` fails due to trailing path separator replaced with `dircat()` instead